app/src/main/java/edu/wisc/ece/pinpoint/pages/pins/PinViewFragment.java

```diff
@@ -88,6 +150,16 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
             // Since pin data shouldn't change, only fetch if not cached
             firebase.fetchPin(pid).addOnCompleteListener(task -> setPinData(task.getResult()));
         }
+
+        fetchAndDisplayComments();
+    }
+
+    private void fetchAndDisplayComments() {
+        firebase.fetchComments(pid).addOnCompleteListener(t -> {
+            comments = (ArrayList<Comment>) t.getResult();
+            commentCount.setText(String.format("comments: %d", comments.size()));
+            setupCommentRecyclerView(getView(), comments);
```

Comment:
See below comment for the comment string. Also see below comments for the recycler view setup & wrapping any `getView()` calls that are in an `onCompleteListener` with a `getView() != null` check.